v2/protocol/kafka_sarama/receiver.go

```diff
@@ -48,7 +50,31 @@ func (r *Receiver) ConsumeClaim(session sarama.ConsumerGroupSession, claim saram
 		m := NewMessageFromConsumerMessage(message)
 
 		r.incoming <- msgErr{
-			msg: binding.WithFinish(m, func(err error) { session.MarkMessage(message, "") }),
+			msg: binding.WithFinish(m, func(err error) {
+				if protocol.IsACK(err) {
```

Comment:
as @slinkydeveloper say, i move the ack/nack logic to binding.WithFinish. then there just 2 files changes from master

Reply:
you mean this? https://github.com/cloudevents/sdk-go/pull/464/files